gammapy/background/reflected.py

```diff
@@ -135,60 +86,92 @@ def __init__(
         else:
-            self.reference_map.data += 1
+            self.reference_map.data = np.ones(self.reference_map.data.shape)
         self.setup()
         self.find_regions()
 
-    @staticmethod
-    def make_reference_map(region, center, binsz="0.02 deg"):
+    def make_reference_map(self):
         """Create empty reference map.
 
         The size of the mask is chosen such that all reflected region are
         contained on the image.
 
-        Parameters
-        ----------
-        region : `~regions.CircleSkyRegion`
-            Region to rotate
-        center : `~astropy.coordinates.SkyCoord`
-            Rotation point
-        binsz : `~astropy.coordinates.Angle`
-            Reference map bin size. Default : 0.02 deg
-
+        Returns
+        -------
+        reference_map : `~gammapy.maps.WcsNDMap`
+            Map containing the region
         """
-        binsz = Angle(binsz)
-        width = 3.0 * region.center.separation(center)
+        width = 15.
+
+        try:
+            reg_center = self.region.center
+        except:
+            raise NotImplementedError("Algorithm not yet adapted to this Region shape")
+
+        if 'ra' in reg_center.representation_component_names:
+            _maskmap = WcsNDMap.create(
+                skydir=self.center, binsz=self.binsz, width=width, coordsys="CEL", proj="TAN"
+             )
+        else:
+            _maskmap = WcsNDMap.create(
+                skydir=self.center, binsz=self.binsz, width=width, coordsys="GAL", proj="TAN"
+            )
+
+        # width is the full width of an image (not the radius)
+        width = Angle(3.0 * reg_center.transform_to(self.center).separation(self.center), u.degree) * 2.
+        maskmap = _maskmap.cutout(self.center, width)
 
-        maskmap = WcsNDMap.create(
-            skydir=center, binsz=binsz, width=width, coordsys="GAL", proj="TAN"
-        )
         return maskmap
 
     def setup(self):
         """Compute parameters for reflected regions algorithm."""
-        wcs = self.reference_map.geom.wcs
-        self._pix_region = self.region.to_pixel(wcs)
-        self._pix_center = PixCoord(*self.center.to_pixel(wcs))
+        geom = self.reference_map.geom
+        self._pix_region = self.region.to_pixel(geom.wcs)
+        self._pix_center = PixCoord.from_sky(self.center, geom.wcs)
         dx = self._pix_region.center.x - self._pix_center.x
         dy = self._pix_region.center.y - self._pix_center.y
 
+        if self._pix_region.contains(self._pix_center):
+            log.warn("Obs #{} rejected! Pointing position within the ON region".format(self.obs_id))
+            return
+
         # Offset of region in pix coordinates
         self._offset = np.hypot(dx, dy)
 
+        # Make the ON reference map
+        _mask = geom.region_mask([self.region], inside=True)
+        self.on_reference_map = WcsNDMap(geom=geom, data=_mask)
+        # TODO: remove this lign after the correction of the Issue #2074
```

Comment:
Resolve this TODO?

Reply:
This ToDo is a reference towards an on-going work, that is not absolutely necessary for this class. It is just a fair mention of possible future code clean-up.
I would not like that this PR is delayed just because of that... This class should be in the next gammapy version.

Reply:
The TODO references issue #2074, but this issue was resolved and closed by @adonath via #2078 on March 28. So these TODOs should be resolved in this PR before merging.

@registerrier - usually what I do is to push commits to a PR branch and then hit the merge button. This is better than merging first and then follow-up commits in master or new PRs, because it's cleaner in the changelog and triggers CI builds in a better way (don't break master).